sadx-onion-blur/sadx-onion-blur.cpp

```diff
@@ -34,81 +34,88 @@ static void __cdecl njAction_Onion(NJS_ACTION* action, float frame)
 			frame = frame_count + frame;
 		}
 
-		njAction_Queue(action, frame, QueuedModelFlagsB_EnableZWrite);
+		late_Action(action, frame, LATE_WZ);
 	}
 
-	RestoreConstantAttr();
-	_nj_control_3d_flag_ = control_3d_orig;
-	GlobalSpriteColor = color_orig;
+	LoadConstantAttr();
+	nj_control_3d_flag_ = control_3d_orig;
+	cur_argb = color_orig;
 }
 
-static const void* loc_494400 = reinterpret_cast<const void*>(0x494400);
+static const void* DrawSonicMotion_loc = reinterpret_cast<const void*>(0x00494400); // DrawSonicMotion usercall
 
-static void __cdecl sub_494400_o(int a1, CharObj2* a2)
+static void __cdecl DrawSonicMotion_o(int a1, playerwk* a2)
 {
 	__asm
 	{
 		mov eax, a1
 		mov esi, a2
-		call loc_494400
+		call DrawSonicMotion_loc
 	}
 }
 
-static void __cdecl sub_494400_c(int anim_index, CharObj2* data2)
+static void __cdecl DrawSonicMotion_c(int anim_index, playerwk* data2)
 {
-	if (MetalSonicFlag || anim_index != 13)
+	if (gu8flgPlayingMetalSonic)
+		goto original;
+	switch (anim_index)
 	{
-		sub_494400_o(anim_index, data2);
+	case 13: // Running
+	case 14: // Spindash
+		ds_ActionClip(data2->mj.plactptr[anim_index].actptr, data2->mj.nframe, 0.0f);
+		njAction_Onion(data2->mj.plactptr[anim_index].actptr, data2->mj.nframe);
 		return;
+	default:
+		goto original;
 	}
-
-	njAction(data2->AnimationThing.AnimData[anim_index].Animation, data2->AnimationThing.Frame);
-	njAction_Onion(data2->AnimationThing.AnimData[anim_index].Animation, data2->AnimationThing.Frame);
+original:
+	DrawSonicMotion_o(anim_index, data2);
+	return;
```

Comment:
This can and probably should be rewritten to avoid the use of goto (wrap the switch statement in `if (!gu8flgPlayingMetalSonic)`, change `default` block to simply `break;`).

Also, preferred style is to indent case from switch statement, i.e.
```cpp
switch (...)
{
case ...:
	do_something();
	break;
```

becomes:
```cpp
switch (...)
{
	case ...:
		do_something();
		break;
```